### PR TITLE
bpo-41816: `StrEnum.__str__` is `str.__str__`

### PR DESCRIPTION
**Pull request by ethanfurman**
use `str.__str__` for StrEnum` so that `str(StrEnum)` is the same as directly accessing the string value of a `StrEnum` member.

https://bugs.python.org/issue41816

### CODE REVIEW

File: Lib/test/test_enum.py [hunk 1]
```diff
@@ -2036,6 +2036,35 @@ class GoodStrEnum(StrEnum):
             two = '2'
             three = b'3', 'ascii'
             four = b'4', 'latin1', 'strict'
+        self.assertEqual(GoodStrEnum.one, '1')
+        self.assertEqual(GoodStrEnum.one, str(GoodStrEnum.one))
```
Comment (serhiy-storchaka) [inline Lib/test/test_enum.py:2041]: Is it what you meant? It does not test that `str(GoodStrEnum.one)` is equal to '1', it tests that it is equal to `GoodStrEnum.one` which is a different thing. `GoodStrEnum.one` may be equal to several different strings. Maybe you meant `self.assertEqual(str(GoodStrEnum.one), '1')`?

I would add also test for `format(GoodStrEnum.one)`.

File: Lib/test/test_enum.py [hunk 2]
```diff
@@ -2036,6 +2036,35 @@ class GoodStrEnum(StrEnum):
             two = '2'
             three = b'3', 'ascii'
```
Comment (serhiy-storchaka) [inline Lib/test/test_enum.py:2037]: I am surprised that it is not an error. Do we have a need to support this syntax?

File: Lib/test/test_enum.py (hunk 1 above)
Comment (ethanfurman) [reply Lib/test/test_enum.py:2041]: It's a two-step process:

- check that `GoodStrEnum.one` is the string `"1"`
- check that `str(GoodStrEnum.one)` is also the string `"1"`

Adding `format()` test.

File: Lib/test/test_enum.py (hunk 2 above)
Comment (ethanfurman) [reply Lib/test/test_enum.py:2037]: It's a standard Enum feature that everything after the `=` is passed to the mixed-in data type's constructor, so whatever it supports is automatically supported.  I just threw in some checks for the other `str` arguments to give friendlier error messages.

File: Lib/test/test_enum.py (hunk 1 above)
Comment (ethanfurman) [reply Lib/test/test_enum.py:2041]: Format test added.

Comment (serhiy-storchaka) [reply Lib/test/test_enum.py:2041]: But from `a == b` and `a == c` it does not follow that `b == c`. Why do not test that `str(GoodStrEnum.one)` equals to string `"1"` directly?

Comment (ethanfurman) [reply Lib/test/test_enum.py:2041]: Okay, direct test added.  I kept the other test as well, since `b == c` should be true in this case.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**